pilot/pkg/networking/core/v1alpha3/route/route_cache.go

```diff
@@ -78,7 +78,7 @@ func (r *Cache) DependentConfigs() []model.ConfigKey {
 	configs := make([]model.ConfigKey, 0, len(r.Services)+len(r.VirtualServices)+
 		len(r.DelegateVirtualServices)+len(r.DestinationRules)+len(r.EnvoyFilterKeys))
 	for _, svc := range r.Services {
-		configs = append(configs, model.ConfigKey{Kind: gvk.ServiceEntry, Name: string(svc.Hostname), Namespace: svc.Attributes.Namespace})
+		configs = append(configs, model.ConfigKey{Kind: gvk.ServiceEntry, Name: svc.Attributes.Namespace, Namespace: svc.Attributes.Namespace})
```

Comment:
`svc.Attributes.Namespace` for `Name` seems wrong. I do think Hostname is actually proper here; ConfigKey ServiceEntry actually uses hostname (which is a bit odd)

Reply:
Yeah sorry, this is a typo.

>  I did think Hostname is actually proper here; ConfigKey ServiceEntry actually uses hostname (which is a bit odd)

Oh.. hmm I see, I guess the cache is hit just because namespace is empty then. Let me update that.